clippy_utils/src/check_proc_macro.rs

```diff
@@ -310,6 +358,18 @@ impl<'cx> WithSearchPat for (&FnKind<'cx>, &Body<'cx>, HirId, Span) {
     }
 }
 
+impl<'cx> WithSearchPat for (&Attribute, &LateContext<'cx>) {
```

Comment:
issue: it doesn't seem like the second half of the tuple is at all necessary

and then we can just use the macro

Reply:
You're correct, `&'cx Attribute` works.

Unfortunately the macro doesn't since `Attribute` has 0 lifetime arguments. `impl_with_search_pat` can likely be modified to only take the lifetime optionally

Reply:
I'm fine with just removing the tuple but keeping the manual impl with a comment about the lifetime and the macro